### PR TITLE
ref(shared-components): refactor tagDistributionMeter to use styled components

### PR DESCRIPTION
**Pull request by Chrissy**
before and after are in the same gif:

(hint: there is no difference)

### CODE REVIEW

File: src/sentry/static/sentry/less/shared-components.less [hunk 1]
```diff
@@ -1151,121 +1151,6 @@ table.integrations {
   width: ((@gridColumnWidth) * @columnSpan) + (@gridGutterWidth * (@columnSpan - 1)) - 10;
 }
 
-// Distribution graph
```
Comment (markstory) [inline src/sentry/static/sentry/less/shared-components.less]: :clap::tada:

**Review by markstory — APPROVED** [review]
Looks good to me.

**Review by billyvg — CHANGES_REQUESTED** [review]
Can we clean up these proptype warnings?

https://travis-ci.org/getsentry/sentry/jobs/467323287#L817

**Comment by Chrissy** [comment]
@billy yees 🙏 🙏🙏🙏

**Comment by Chrissy** [comment]
@billyvg should be fixed

**Comment by Chrissy** [comment]
looks like i'm still missing some linter stuff one sec...

File: src/sentry/static/sentry/app/components/group/tagDistributionMeter.jsx [hunk 2]
```diff
@@ -159,15 +164,94 @@ const TagDistributionMeter = createReactClass({
...
+const Tag = styled('div')`
+  position: relative;
+  font-size: 13px;
+  margin: 10px 0 8px;
+  font-weight: bold;
+  z-index: 5;
+  line-height: 1;
+`;
+
+const Description = styled('span', {shouldForwardProp: p => isPropValid(p)})`
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/group/tagDistributionMeter.jsx]: I think the `isPropValid` fat arrow function here is the default for shouldForwardProp. You would only need to do this if you were passing in some props that are usually passed to the DOM and then excluding them

eg

```js
{shouldForwardProp: p => p !== 'disabled' && isPropValid(p)}
```

Comment (Chrissy) [reply src/sentry/static/sentry/app/components/group/tagDistributionMeter.jsx]: hmm. I guess I'm not sure how to implement the default then? It is for sure passing `first` and `last` to the underlying html element when I only do `styled('div')`. When I add `isPropValid` like the above it then filters out `first` and `last`.

Comment (Chrissy) [reply src/sentry/static/sentry/app/components/group/tagDistributionMeter.jsx]: something like `shouldForwardProp: isPropValid`?

**Comment by Chrissy** [comment]
alrighty everything is passing now

**Review by billyvg — APPROVED** [review]

**Pull request merged**